File: Taller de Haskell/Problema_9/problema_9.hs
-- Importamos funciones necesarias para entrada/salida y validación
import System.IO (hFlush, stdout)
import Data.Char (toUpper)
import Text.Read (readMaybe)

-- | 'leerOpcionSN' solicita una opción de tipo S/N (Sí/No) y la convierte en mayúsculas
leerOpcionSN :: String -> IO (Maybe Char)
leerOpcionSN mensaje = do
    putStr mensaje
    hFlush stdout
    entrada <- getLine
    case map toUpper entrada of
        "S" -> return (Just 'S')
        "N" -> return (Just 'N')
        _   -> return Nothing

-- | Lista de ingredientes base para pizza vegana
ingredientesVeganos :: [String]
ingredientesVeganos =
    [ "1. Champiñones"
    , "2. Pimientos asados"
    , "3. Aceitunas negras"
    , "4. Cebolla caramelizada"
    , "5. Queso vegano"
    ]

-- | Lista de ingredientes para pizza normal
ingredientesNormales :: [String]
ingredientesNormales =
    [ "1. Queso"
    , "2. Pepperoni"
    , "3. Jamón"
    , "4. Champiñones"
    , "5. Pimientos"
    ]

-- | Función principal que guía la selección de pizza
main :: IO ()
main = do
    opcionPizza <- leerOpcionSN "¿Pizza vegana? (S/N): "
    case opcionPizza of
        Just 'S' -> do
            putStrLn "Ingredientes base de la pizza vegana:"
            mapM_ putStrLn ingredientesVeganos

            extra <- leerOpcionSN "¿Quieres albahaca fresca como ingrediente extra? (S/N): "
            case extra of
                Just 'S' -> do
                    putStrLn "\nLa pizza vegana que pediste tiene:"
                    mapM_ putStrLn (ingredientesVeganos ++ ["6. Albahaca fresca"])
                Just 'N' -> do
                    putStrLn "\nLa pizza vegana que pediste tiene:"
                    mapM_ putStrLn ingredientesVeganos
                Nothing  -> putStrLn "Opción inválida para ingrediente extra. Por favor responde con 'S' o 'N'."
        Just 'N' -> do
            putStrLn "La pizza normal que pediste tiene:"
            mapM_ putStrLn ingredientesNormales
        Nothing -> putStrLn "Opción inválida. Por favor responde con 'S' para pizza vegana o 'N' para pizza normal."

File: Taller de Haskell/Problema_9/problema_9.py
# Solicita al usuario si desea una pizza vegana
opcion = input("¿Pizza vegana? (S/N): ").upper()
# Verifica si la opción es vegana
if opcion == "S":
    print("Ingredientes base de la pizza vegana:")
    print("1. Champiñones")
    print("2. Pimientos asados")
    print("3. Aceitunas negras")
    print("4. Cebolla caramelizada")
    print("5. Queso vegano")
    # Pregunta si desea albahaca fresca como ingrediente extra
    extra = input("¿Quieres albahaca fresca como ingrediente extra? (S/N): ").upper()
    # Verifica si el usuario quiere albahaca fresca
    if extra == "S":
        print("\nLa pizza vegana que pediste tiene:")
        print("1. Champiñones")
        print("2. Pimientos asados")
        print("3. Aceitunas negras")
        print("4. Cebolla caramelizada")
        print("5. Queso vegano")
        print("6. Albahaca fresca")
    elif extra == "N":
        print("\nLa pizza vegana que pediste tiene:")
        print("1. Champiñones")
        print("2. Pimientos asados")
        print("3. Aceitunas negras")
        print("4. Cebolla caramelizada")
        print("5. Queso vegano")
    else:
        print("Opción inválida para ingrediente extra. Por favor responde con 'S' o 'N'.")
# Verifica si la opción es pizza normal
elif opcion == "N":
    print("La pizza normal que pediste tiene:")
    print("1. Queso")
    print("2. Pepperoni")
    print("3. Jamón")
    print("4. Champiñones")
    print("5. Pimientos")
# Manejo de errores si la entrada no es válida
else:
    print("Opción inválida. Por favor responde con 'S' para pizza vegana o 'N' para pizza normal.")
